stdlib/lang-parser/parserModule/parser.bal

```diff
@@ -0,0 +1,1260 @@
+                DefinitionNode defNode = function1;
+                dList[pos] = defNode;
+                pos += 1;
+            }
+        }
+        //consume the EOF Token
+        Token currToken = self.parserBuffer.consumeToken();
+        PackageNode pkNode = {
+            nodeKind: PACKAGE_NODE,
+            tokenList: [currToken],
+            definitionList: <DefinitionNode[]>(dList.freeze())
+        };
+        return pkNode;
+    }
+
+    #method to consume the token from the token buffer if the token matches the expected token
+    #if the token doesnt match , panic recovery method will be invoked.
+    # +return - Token
+    function matchToken(int mToken, NodeKind rule) returns Token {
+        if (self.LAToken(1) == mToken) {
+            Token currToken = self.parserBuffer.consumeToken();
+            return currToken;
+        } else {
+            if (self.LAToken(1) == LEXER_ERROR_TOKEN) {
+                log:printError(self.LookaheadToken(1).lineNumber + ":" + self.LookaheadToken(1).startPos + ": expected " +
+                tokenNames[mToken] + ";found an invalid token '" + self.LookaheadToken(1).text + "'");
+            } else {
+                log:printError(self.LookaheadToken(1).lineNumber + ":" + self.LookaheadToken(1).startPos + ": expected " +
+                tokenNames[mToken] + "; found '" + self.LookaheadToken(1).text + "'");
+            }
+
+            Token panicToken = self.panicRecovery(mToken, rule);
+            return panicToken;
+        }
+    }
+
+    #error recovery : token  insertion
+    # the expected token will be inserted inplace of the mismatched token
+    # +return - Token
+    function insertToken(int mToken) returns Token {
+        log:printError(tokenNames[mToken] + " inserted");
+        return {
+            tokenType: mToken,
+            text: tokenNames[mToken],
+            startPos: -1,
+            endPos: -1,
+            lineNumber: 0,
+            index: -1,
+            whiteSpace: ""
+        };
+    }
+
+    #error recovery : delete token
+    # mismatched token is consumed and added to the error token list
+    # +return - Token
+    function deleteToken() returns Token {
+        Token invalidToken = self.parserBuffer.consumeToken();
+        log:printError(invalidToken.lineNumber + ":" + invalidToken.startPos + ": invalid token '" +
+        invalidToken.text + "'");
+        self.errTokens[self.errCount] = invalidToken;
+        self.errCount += 1;
+        self.invalidOccurence = true;
+        return invalidToken;
+    }
+
+    #panic recovery
+    #each grammar rule will have a specific error recovery.
+    #here all the token will be removed and added to the error token list until we meet a terminal token
+    # +return - Token
+    function panicRecovery(int mToken, NodeKind rule) returns Token {
+        boolean panicMode = true;
+        self.errorRecovered = false;
+
+        if (rule == "variableDefinitionStatement") {
```

Comment:
We have a constant for "variableDefinitionStatement" right? Shall we use that? Please check other places too.